src/EventStore.Core/Index/PTable.cs

```diff
@@ -767,6 +795,252 @@ public partial class PTable : ISearchTable, IDisposable {
 				throw new TimeoutException();
 		}
 
+		public List<IndexEntry> GetRangeWithCache(StreamHash stream, long startNumber, long endNumber, int? limit = null) {
+			if (!OverlapsRange(stream, startNumber, endNumber, out var tableLatestNumber, out var tableLatestOffset))
+				return new List<IndexEntry>();
+
+			// it does overlap.
+			// if the requested end version is greater than or equal to what we have in this ptable
+			// then we can jump to tableLatestOffset and read the file forwards from there without binary chopping.
+			if (endNumber >= tableLatestNumber) {
+				return PositionAndReadForward(stream, startNumber, endNumber, limit, tableLatestOffset: tableLatestOffset);
+			}
+			// todo: else if the requested start version is less than or equal to what we have in this ptable
```

Comment:
@timothycoleman Just confirming if this TODO is not planned to be done as part of this PR.

Reply:
confirmed